src/duct/middleware/web.clj

```diff
@@ -12,23 +12,25 @@
 (def ^:private request-log-keys
   [:request-method :uri :query-string])
 
-(defn- log-request [logger request]
-  (logger/log logger :info ::request (select-keys request request-log-keys)))
+(defn- log-request [logger request level]
+  (logger/log logger level ::request (select-keys request request-log-keys)))
 
 (defn- log-error [logger ex]
   (logger/log logger :error ::handler-error ex))
 
 (defn wrap-log-requests
   "Log each request using the supplied logger. The logger must implement the
   duct.core.protocols/Logger protocol."
-  [handler logger]
-  (fn
-    ([request]
-     (log-request logger request)
-     (handler request))
-    ([request respond raise]
-     (log-request logger request)
-     (handler request respond raise))))
+  ([handler logger]
+   (wrap-log-requests handler logger :info))
+  ([handler logger level]
```

Comment:
I was thinking that to future-proof this API, it would make sense to make the third argument an options map. So instead of `level`, change it to `{:keys [level] :or {level :info}}`. Then in the `::log-requests` method we can pass the options directly, minus the logger: `(dissoc options :logger)`.

Reply:
Thanks for you reply (and also contributing to Ring. We all clojurian thank a lot for your work).

I pushed new commit with your feedback.
I added third parameter to `wrap-log-requests` function as an option map.
And I revised `test-wrap-log-requests` to add a third parameter.
(We always call `wrap-log-requests` with three parameters. Is it correct?)